File: app/src/main/java/com/yawar/memo/repositry/UserInformationRepo.java
package com.yawar.memo.repositry;

import android.annotation.SuppressLint;
import android.app.Application;
import android.util.Log;

import androidx.lifecycle.MutableLiveData;

import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.yawar.memo.constant.AllConstants;
import com.yawar.memo.model.ChatRoomModel;
import com.yawar.memo.model.MediaModel;
import com.yawar.memo.model.UserModel;
import com.yawar.memo.retrofit.RetrofitClient;
import com.yawar.memo.utils.BaseApp;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

import io.reactivex.Single;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class UserInformationRepo {
    public MutableLiveData<String> blockedFor;
     public MutableLiveData<ArrayList<MediaModel>> mediaModelsMutableLiveData;
    private ArrayList<MediaModel> mediaModels;
    public MutableLiveData<UserModel> userInformation;




//    ChatRoomRepo chatRoomRepo = BaseApp.getInstance().getChatRoomRepo();


    public MutableLiveData<Boolean> blocked;
    public MutableLiveData<Boolean> unBlocked;
    BaseApp myBase = BaseApp.getInstance();




    public UserInformationRepo(Application application) { //application is subclass of context

        mediaModelsMutableLiveData = new MutableLiveData<>();
        mediaModels = new ArrayList<>();
        userInformation = new MutableLiveData<>();

    }





    @SuppressLint("CheckResult")
    public void getMedia(String user_id , String anthor_user_id) {
        mediaModels.clear();


        Single<String> observable = RetrofitClient.getInstance().getapi().getMedia(user_id,anthor_user_id)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread());
        observable.subscribe(s -> {
                      try {

                    JSONArray jsonArray = new JSONArray(s);

                    for (int i = 0; i <= jsonArray.length() - 1; i++) {
                        JSONObject jsonObject = jsonArray.getJSONObject(i);
                        String image = jsonObject.getString("message");
                        mediaModels.add(new MediaModel(image));

                    }
                    mediaModelsMutableLiveData.setValue(mediaModels);


                } catch (JSONException e) {
                    e.printStackTrace();
                }



                },
                s -> {

                    mediaModelsMutableLiveData.setValue(null);
                });
    }

    @SuppressLint("CheckResult")
    public MutableLiveData<UserModel> getUserInformation(String anthor_user_id) {


        Single<String> observable = RetrofitClient.getInstance().getapi().getUserInformation(anthor_user_id)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread());
        observable.subscribe(s -> {
            System.out.println("s"+s);
//            return userInformation;
                    try {

                        JSONArray jsonArray = new JSONArray(s);
                        JSONObject jsonObject = jsonArray.getJSONObject(0);
                        UserModel userModel = new UserModel(jsonObject.getString("id"),jsonObject.getString("first_name"),jsonObject.getString("last_name"),jsonObject.getString("email"),jsonObject.getString("phone"),jsonObject.getString("sn"),jsonObject.getString("profile_image"),jsonObject.getString("status"));


                     userInformation.setValue(userModel);


                    } catch (JSONException e) {
                        e.printStackTrace();
                    }



                },
                s -> {
                    userInformation.setValue(null);
//                    mediaModelsMutableLiveData.setValue(null);
                });
        return userInformation;
    }


}

File: app/src/main/java/com/yawar/memo/repositry/UserInformationRepo.kt
package com.yawar.memo.repositry

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.yawar.memo.Api.GdgApi
import com.yawar.memo.constant.AllConstants
import com.yawar.memo.model.MediaModel
import com.yawar.memo.model.UserModel
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.Job
import kotlinx.coroutines.launch
import org.json.JSONArray
import org.json.JSONObject

class UserInformationRepo {
    private val _mediaModelsMutableLiveData = MutableLiveData<ArrayList<MediaModel>>()
    val mediaModelsMutableLiveData: LiveData<ArrayList<MediaModel>>
        get() = _mediaModelsMutableLiveData

    private val _blockedFor =  MutableLiveData<String>()
    val blockedFor : LiveData<String>
        get() = _blockedFor

    private val _userInformation = MutableLiveData<UserModel>()
    val userInformation: LiveData<UserModel>
        get() = _userInformation

    private val _blocked =  MutableLiveData<Boolean>()
    val blocked : LiveData<Boolean>
        get() = _blocked

    private var viewModelJob = Job()
    private val coroutineScope = CoroutineScope(viewModelJob + Dispatchers.Main )

    fun  getMedia( user_id : String ,  anthor_user_id :String){
        var mediaModels = ArrayList<MediaModel>()
        _mediaModelsMutableLiveData.value = ArrayList()
        coroutineScope.launch {

//            val getMediaDeferred = GdgApi(AllConstants.base_node_url).apiService
//                .getMedia(user_id,anthor_user_id)
            val getMediaDeferred = GdgApi.apiService
                .getMedia(user_id,anthor_user_id)

            try {
                val listResult = getMediaDeferred?.await()

                val jsonArray = JSONArray(listResult)

                for (i in 0 until jsonArray.length()) {
                    val jsonObject: JSONObject = jsonArray.getJSONObject(i)
                    val image = jsonObject.getString("message")
                    mediaModels.add(MediaModel(image))
                }
                _mediaModelsMutableLiveData.value = mediaModels


            } catch (e: Exception) {

                Log.d("getMarsRealEstateProperties: ","Failure: ${e.message}")

            }
        }

    }

    fun  getUserInformation( anthor_user_id :String) {
        var mediaModels = ArrayList<MediaModel>()
        _userInformation.value = UserModel()

        coroutineScope.launch {

//            val getMediaDeferred =  GdgApi(AllConstants.base_node_url).apiService
//                .getUserInformation(anthor_user_id)
            val getMediaDeferred =  GdgApi.apiService
                .getUserInformation(anthor_user_id)
            Log.d("getUserInformation", "getUserInformation: ")

            try {
                val listResult = getMediaDeferred?.await()

                Log.d("getUserInformation", listResult.toString())

                val jsonArray = JSONArray(listResult)
                val jsonObject: JSONObject = jsonArray.getJSONObject(0)
                val userModel = UserModel(
                    jsonObject.getString("id"),
                    jsonObject.getString("first_name"),
                    jsonObject.getString("last_name"),
                    jsonObject.getString("email"),
                    jsonObject.getString("phone"),
                    jsonObject.getString("sn"),
                    jsonObject.getString("profile_image"),
                    jsonObject.getString("status")
                )
                _userInformation.value = userModel


            } catch (e: Exception) {

                Log.d("getMarsRealEstateProperties: ", "Failure: ${e.message}")

            }
        }

    }

}